taskcluster/mozillavpn_taskgraph/transforms/multi_dep.py

```diff
@@ -1,122 +0,0 @@
-# License, v. 2.0. If a copy of the MPL was not distributed with this
-# file, You can obtain one at http://mozilla.org/MPL/2.0/.
-"""
-Apply some defaults and minor modifications to the single_dep jobs.
-"""
-
-
-from taskgraph.transforms.base import TransformSequence
-from taskgraph.util.schema import resolve_keyed_by
-from taskgraph.util.treeherder import inherit_treeherder_from_dep, join_symbol
-
-
-transforms = TransformSequence()
-
-
-@transforms.add
-def build_name_and_attributes(config, tasks):
-    for task in tasks:
-        task["dependencies"] = {
-            dep_key: dep.label for dep_key, dep in _get_all_deps(task).items()
-        }
-        primary_dep = task["primary-dependency"]
-        copy_of_attributes = primary_dep.attributes.copy()
-        task.setdefault("attributes", copy_of_attributes)
-        # run_on_tasks_for is set as an attribute later in the pipeline
-        task.setdefault("run-on-tasks-for", copy_of_attributes["run_on_tasks_for"])
-        task["name"] = _get_dependent_job_name_without_its_kind(primary_dep)
-
-        yield task
-
-
-def _get_dependent_job_name_without_its_kind(dependent_job):
-    return dependent_job.label[len(dependent_job.kind) + 1 :]
-
-
-def _get_all_deps(task):
-    if task.get("dependent-tasks"):
-        return task["dependent-tasks"]
-
-    return {task["primary-dependency"].kind: task["primary-dependency"]}
-
-
-@transforms.add
-def resolve_keys(config, tasks):
-    for task in tasks:
-        resolve_keyed_by(
-            task,
-            "treeherder.job-symbol",
-            item_name=task["name"],
-            **{
-                "build-type": task["attributes"]["build-type"],
-                "level": config.params["level"],
-            },
-        )
-        yield task
-
-
-@transforms.add
-def resolve_keys(config, tasks):
-    for task in tasks:
-        resolve_keyed_by(
-            task,
-            "treeherder.platform",
-            item_name=task["name"],
-            **{
-                "build-type": task["attributes"]["build-type"],
-                "level": config.params["level"],
-            },
-        )
-        yield task
-
-
-@transforms.add
-def build_upstream_artifacts(config, tasks):
-    for task in tasks:
-        worker_definition = {
-            "upstream-artifacts": [],
-        }
-
-        for dep in _get_all_deps(task).values():
-            paths = sorted(
-                artifact["name"]
-                for artifact in dep.attributes.get("release-artifacts", [])
-            )
-
-            if paths:
-                worker_definition["upstream-artifacts"] = [
-                    {
-                        "taskId": {"task-reference": f"<{dep.kind}>"},
-                        "taskType": dep.kind,
-                        "paths": paths,
-                    }
-                ]
-
-        task.setdefault("worker", {}).update(worker_definition)
-        yield task
-
-
-@transforms.add
-def build_treeherder_definition(config, tasks):
```

Comment:
I had no idea treeherder things were in this transform (and why would I based on the name...). Glad to see this getting reorganized!